### DISCUSSION TITLE
BLE Device COM Selection

### DISCUSSION THREAD

**Discussion by LimeSlice** [hbldh/bleak#835] [Ideas]
I haven't seen it already implemented, but are there any plans to introduce a COM port keyword argument into the init of the BleakClient and BleakScanner?

**Comment by dlech** [comment]
Bleak is Bluetooth Low Energy only, which doesn't include Bluetooth serial ports from Bluetooth Classic. Something like PySerial would be better suited to using Bluetooth as a COM port on Windows.

**Reply by LimeSlice** [reply]
So I'm using a Nordic evaluation board that has a NRF52 BLE module attached. I've previously been using the pc_ble_driver_py library for connecting from the NRF52 to my devices. I was interested in Bleak because it's a lot simpler, but in pc_ble_driver_py, I would specify the COM port the NRF52 board was connected to so that if I had multiple COM ports attached, it knew exactly which port to connect through.

Is this what you are talking about when you are mentioning Bluetooth serial ports?

**Reply by dlech** [reply]
The Nordic UART Service on it's own doesn't have a notion of COM ports, so I assumed that you meant COM ports as in serial ports on Windows. Perhaps this `pb_ble_drive_py` tool is using the "COM" terminology for something else?
